cnb_image.go

```diff
@@ -26,6 +26,8 @@ type CNBImageCore struct {
 	preferredMediaTypes MediaTypes
 	preserveHistory     bool
 	previousImage       v1.Image
+	features            []string
+	annotations         map[string]string
```

Comment:
Why do these have to be fields on the struct? Could they not live on the manifest and config?

Reply:
these fields exists in [v1.Manifest](https://github.com/google/go-containerregistry/blob/051d642e295e3e86b58bb5feabb184036fa34a52/pkg/v1/manifest.go#L25-L53) but not in [v1.ConfigFile](https://github.com/google/go-containerregistry/blob/051d642e295e3e86b58bb5feabb184036fa34a52/pkg/v1/config.go#L29-L42), 
If these fields should be removed then, I think we should change the code as following, so that mutating `os`, `arch`, `variant`, `osVersion`, `osFeatures` should reflect changes in both ConfigFile and v1.Manifest
```go
	func (i *CNBImageCore) MutateConfigFile(withFunc func(c *v1.ConfigFile)) error {
	// FIXME: put MutateConfigFile on the interface when `remote` and `layout` packages also support it.
	configFile, err := getConfigFile(i.Image)
	if err != nil {
		return err
	}
	withFunc(configFile)
	i.Image, err = mutate.ConfigFile(i.Image, configFile)
	return MutateManifest(.....)
}
```

Reply:
I am not sure that you need to explicitly call MutateManifest - the image that you get back from calling `mutate.ConfigFile` is going to re-compute the manifest when you call `image.Manifest()` or whatever. Unless I'm misunderstanding something. I don't see `arch`, `variant`, etc. in the [manifest spec](https://github.com/opencontainers/image-spec/blob/main/manifest.md) so I assume you mean re-computing the config digest in the manifest which should be covered.

Reply:
So looking at GGCR, I think we are good - the index will compute the descriptor from the added image; as long as the image has the right platform, this will bubble up to the index: https://github.com/google/go-containerregistry/blob/051d642e295e3e86b58bb5feabb184036fa34a52/pkg/v1/mutate/index.go#L123

Reply:
I was not aware of [manifest spec](https://github.com/opencontainers/image-spec/blob/main/manifest.md) not having `os`, `arch`, `variant` etc.., field 

I think we can now safely delete `MutateManifest` calls for `CNBImage` but at the same time we should also change getters and setters of `CNBImage` `Features` to process `mfest.Subject` instead of `mfest.Config` or should we return nil?

And I also think that this change will also cleanup a lot of code from this PR(as I thought to add `os`, `arch`, `variant` which are not getting added by GGCR which significantly increased PR code size)

Reply:
I am not sure that we need to worry about manifest.Subject for the time being. Looking at the spec:

> subject [descriptor](https://github.com/opencontainers/image-spec/blob/main/descriptor.md)

> This OPTIONAL property specifies a [descriptor](https://github.com/opencontainers/image-spec/blob/main/descriptor.md) of another manifest. This value defines a weak association to a separate [Merkle Directed Acyclic Graph (DAG)](https://en.wikipedia.org/wiki/Merkle_tree) structure, and is used by the [referrers API](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers) to include this manifest in the list of responses for the subject digest.

I believe this is when the index in question is an aggregate of SBOMs or signatures relating to another image. Since we are just publishing regular images we shouldn't need it.

Reply:
Will be resolved by https://github.com/husni-faiz/imgutil/pull/5